### PR TITLE
8344397: Remove Security Manager dependencies from java.security and sun.security packages

### PR DESCRIPTION
**Pull request by seanjmullan**
Now that JEP 486 has been integrated, `java.security` and `sun.security` implementation dependencies on `System.getSecurityManager` and `AccessController.doPrivileged` can be removed. 

This should cover most of the remaining cleanup in the security libraries area. A couple of the `sun.security.action` classes can't be removed yet, as well as a few constants in `sun.security.util.SecurityConstants` and internal `ProtectionDomain` methods until dependencies are removed in other areas. Some additional test cleanup can also be done later.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344397](https://bugs.openjdk.org/browse/JDK-8344397): Remove Security Manager dependencies from java.security and sun.security packages (**Enhancement** - P3)

### Reviewers
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**) Review applies to [fdc7dfd3](https://git.openjdk.org/jdk/pull/22418/files/fdc7dfd353a925fbf1eb0e28595dfca88ff4076e)
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Hai-May Chao](https://openjdk.org/census#hchao) (@haimaychao - Committer)
 * [Weijun Wang](https://openjdk.org/census#weijun) (@wangweij - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22418/head:pull/22418` \
`$ git checkout pull/22418`

Update a local copy of the PR: \
`$ git checkout pull/22418` \
`$ git pull https://git.openjdk.org/jdk.git pull/22418/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22418`

View PR using the GUI difftool: \
`$ git pr show -t 22418`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22418.diff">https://git.openjdk.org/jdk/pull/22418.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22418#issuecomment-2504489280)
</details>

### CODE REVIEW

**Review by AlanBateman — APPROVED** [review]
I did a pass over this and didn't spot any issues.

**Review by RogerRiggs — APPROVED** [review]
lgtm

**Review by haimaychao — APPROVED** [review]

**Comment by haimaychao** [comment]
Looks good.

File: src/java.base/share/classes/java/security/Provider.java
Comment (franferrax) [inline src/java.base/share/classes/java/security/Provider.java:1]: Hi @seanjmullan,

This is very minor/nit, but, while adapting #15539 for the upcoming changes in this PR (and the whole work in [JDK-8338411](https://bugs.openjdk.org/browse/JDK-8338411 "Implement JEP 486: Permanently Disable the Security Manager")), I found the following comments, which we will also want to remove.

https://github.com/openjdk/jdk/blob/8abeb41b740534e832135a9e29f53460e4c509f2/src/java.base/share/classes/java/security/Provider.java#L348-L352
https://github.com/openjdk/jdk/blob/8abeb41b740534e832135a9e29f53460e4c509f2/src/java.base/share/classes/java/security/Provider.java#L779-L780

File: src/java.base/share/classes/sun/security/util/SecurityConstants.java [hunk 1]
```diff
@@ -63,76 +58,13 @@ private SecurityConstants () {
 
     // Permission constants used in the various checkPermission() calls in JDK.
 
-    // java.lang.Class, java.lang.SecurityManager, java.lang.System,
-    // java.net.URLConnection, java.security.AllPermission, java.security.Policy,
-    // sun.security.provider.PolicyFile
+    // java.lang.Class, java.net.URLConnection, java.security.AllPermission
```
Comment (AlanBateman) [inline src/java.base/share/classes/sun/security/util/SecurityConstants.java:61]: BTW: It's okay to change the usage in Class.Holder.<clinit> to use `new AllPermission()`. That will help get to the point where this file can be removed.

**Comment by dfuch** [comment]
I am seing some doPrivileged in `src/java.base/unix/classes/sun/security/provider/NativePRNG.java` - should this PR also fix them?

**Review by wangweij — COMMENTED** [review]
`OCSP.java` and `URICertStore` in `sun/security/provider/certpath` still using `GetPropertyAction`.

File: src/java.base/share/classes/java/security/Provider.java
Comment (seanjmullan) [reply src/java.base/share/classes/java/security/Provider.java:1]: Good catch. Also noticed a couple of other comments with "permission" and removed that text. Fixed in https://github.com/openjdk/jdk/pull/22418/commits/283fb2c12cbfb3a34a22fa3760d0879767ceefe0

File: src/java.base/share/classes/sun/security/util/SecurityConstants.java (hunk 1 above)
Comment (seanjmullan) [reply src/java.base/share/classes/sun/security/util/SecurityConstants.java]: Fixed in https://github.com/openjdk/jdk/pull/22418/commits/5ad5e8c8b13e3aa8f8379def6b3bb3e994fd084d.

**Review by AlanBateman — APPROVED** [review]

File: src/java.base/share/classes/java/security/Provider.java
[the remark this answers is not in the document]
Comment (seanjmullan) [reply src/java.base/share/classes/java/security/Provider.java:1]: Fixed in https://github.com/openjdk/jdk/pull/22418/commits/283fb2c12cbfb3a34a22fa3760d0879767ceefe0 and https://github.com/openjdk/jdk/pull/22418/commits/fdc7dfd353a925fbf1eb0e28595dfca88ff4076e.

**Review by RogerRiggs — APPROVED** [review]

**Review by haimaychao — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]

**Review by haimaychao — APPROVED** [review]

**Review by wangweij — APPROVED** [review]

**Pull request closed without merging**